src/Sylius/Bundle/ApiBundle/Controller/Payment/GetPaymentConfiguration.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Controller\Payment;
+
+use Sylius\Bundle\ApiBundle\Provider\PaymentConfigurationProviderInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\PaymentInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Sylius\Component\Core\Repository\PaymentRepositoryInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Webmozart\Assert\Assert;
+
+/** @experimental */
+final class GetPaymentConfiguration
+{
+    /** @var PaymentRepositoryInterface */
+    private $paymentRepository;
+
+    /** @var PaymentConfigurationProviderInterface */
+    private $paymentConfigurationProvider;
+
+    /** @var OrderRepositoryInterface $orderRepository */
+    private $orderRepository;
+
+    public function __construct(
+        PaymentRepositoryInterface $paymentRepository,
+        PaymentConfigurationProviderInterface $paymentConfigurationProvider,
+        OrderRepositoryInterface $orderRepository
+    ) {
+        $this->paymentRepository = $paymentRepository;
+        $this->paymentConfigurationProvider = $paymentConfigurationProvider;
+        $this->orderRepository = $orderRepository;
+    }
+
+    public function __invoke(Request $request): JsonResponse
+    {
+        /** @var PaymentInterface|null $payment */
+        $payment = $this->paymentRepository->find($request->get('paymentId'));
+
+        /** @var OrderInterface|null $order */
+        $order = $this->orderRepository->findOneByTokenValue($request->get('id'));
+
+        Assert::notNull($order);
+        Assert::notNull($payment);
```

Comment:
I would move `Assert::notNull($payment);` just under `$payment = $this->paymentRepository->find($request->get('paymentId'));` so in case $payment is null we want call DB second time. In other way grouped asserts looks nicer so it is up to you 😄

Reply:
you're right, but its cached and i dont think its a big deal tho. Code looks better this way i'd say

Reply:
I would be even for only one call to DB to get payment for given id and order token value. Or you can get order firstly, and then get the payment for id and order, in this case, there is no need for custom repository method, but there will be 2 calls to DB.

Reply:
1 call to the database based on order token and paymentId FTW